### PR TITLE
GalSim: fix build when homebrew is not in /usr/local or other default search path

### PR DESCRIPTION
**Pull request by dstndstn**
As reported in #2577 ; this version is reported to work by OP

### CODE REVIEW

File: gal-sim.rb [hunk 1]
```diff
@@ -30,7 +30,7 @@ def install
       end
       args << "WITH_OPENMP=true"
     end
-    scons *args
+    scons "EXTRA_INCLUDE_PATH=#{HOMEBREW_PREFIX}/include", "EXTRA_LIB_PATH=#{HOMEBREW_PREFIX}/lib", "EXTRA_PATH=#{HOMEBREW_PREFIX}/bin", *args
```
Comment (dpo) [inline gal-sim.rb]: Why not just add `Formula["fftw3"].opt_include`, etc., instead of `HOMEBREW_PREFIX`?

**Comment by dstndstn** [comment]
That might be an option.  There are scons variables for FFTW, TMV, and Boost paths.  Presumably scons can find the correct python include path.  (And the fftw path should be found with pkg-config anyway.)  I'm not sure I prefer this, though -- since *all* the homebrew headers are installed in a nonstandard place, to me it makes sense to tell scons that.  That seems preferable to telling it that FFTW, TMV, and Boost are each to be found in /path/to/homebrew.

(Also, I don't have a strange homebrew install like this for testing, and I'm not sure the OP is available for testing.  But that's just an excuse!)

thx

**Comment by dpo** [comment]
There's a slight inconsistency about this formula. The formula name is `GalSim` which means that the destination folder in the cellar is `gal-sim`. Yet `scons install` installs stuff in `share/galsim`, not in `share/gal-sim`. The [Github site](https://github.com/GalSim-developers/GalSim) is called `GalSim`, not `Gal-Sim`, so shouldn't we really rename this formula `Galsim` and the formula file `galsim.rb`.

If I try to install the example scripts (which aren't installed by default) using `pkgshare.install`, they end up in `share/gal-sim`, so we have both `share/galsim` and `share/gal-sim`...

File: gal-sim.rb (hunk 1 above)
Comment (dpo) [inline gal-sim.rb]: This works for me: https://gist.github.com/dpo/931cc097d7385194edac
I also added tests and installed the examples.

**Comment by iml** [comment]
Might want to wait for the formula renames work to be merged before renaming, otherwise it will strand users with the old formula installed. Could use `(share/"galsim")` in the meantime.

**Comment by dstndstn** [comment]
Hmm, I wanted to call the formula galsim when I submitted it, but somehow I
thought the naming rules were such that CamelCase names got converted to
dashes.  I no longer see a rule like that, so maybe I was mistaken.  Also
"brew create" with the download URL creates a formula named galsim.rb.  I
would suggest that you submit a pull request to make that name change :)

**Pull request closed without merging**